lib/services/v2.0/kc2_service_interface.dart

```diff
@@ -0,0 +1,473 @@
+      rethrow;
+    }
+  }
+
+  Future<List<Transaction>> getTransactionsByPhoneNumberHash(
+      String phoneNumberHash) async {
+    try {
+      // Cut `0x` prefix if exists
+      if (phoneNumberHash.startsWith('0x')) {
+        phoneNumberHash = phoneNumberHash.substring(2);
+      }
+
+      List<dynamic> result = await callRpc(
+          'transactions_getTransactionsByPhoneNumberHash', [phoneNumberHash]);
+      return result.map((e) => Transaction.fromJson(e)).toList();
+    } on PlatformException catch (e) {
+      debugPrint(
+          'Failed to get transactions_getTransactionsByPhoneNumberHash: ${e.message}');
+      rethrow;
+    }
+  }
+
+  getVerificationEvidence(String accountId, String username, String phoneNumber,
+      {String? byPassToken}) async {
+    try {
+      Map<String, dynamic>? result = await callRpc(
+          'verifier_verify', [accountId, username, phoneNumber, byPassToken]);
+
+      return result == null ? null : VerificationEvidence.fromJson(result);
+    } on PlatformException catch (e) {
+      debugPrint('Failed to get verification evidence: ${e.message}');
+      rethrow;
+    }
+  }
+
+  // transactions
+
+  /// Create a new on-chain user with provided data
+  /// accountId - ss58 encoded user's public ed25519 key
+  /// userName - unique username. Must not be empty
+  /// phoneNumber - user's phone number. Including country code. Excluding leading +
+  /// Returns an (evidence, errorMessage) result.
+  /// @HolyGrease - first string in tuple should be the tx hash and not evidence!
+  ///
+  /// This method will attempt to obtain verifier evidence regarding the association between the accountId, userName and phoneNumber
+  Future<(String?, String?)> newUser(
+      String accountId, String username, String phoneNumber,
+      {VerificationEvidence? verificationEvidence}) async {
+    try {
+      // Get verification evidence if not provided
+      verificationEvidence ??= await getVerificationEvidence(
+          accountId, username, phoneNumber,
+          byPassToken: verificationBypassToken);
+
+      // Failed to get verification evidence
+      if (verificationEvidence == null) {
+        return (null, 'NoVerifierEvidence');
+      }
+
+      // Verification failed
+      if (verificationEvidence.verificationResult !=
+          VerificationResult.verified) {
+        return (null, verificationEvidence.verificationResult.toString());
+      }
+
+      final phoneNumberHash = hasher.hashString(phoneNumber);
+      final call = MapEntry(
+          'Identity',
+          MapEntry('new_user', {
+            'verifier_public_key':
+                decodeAccountId(verificationEvidence.verifierAccountId!),
+            'verifier_signature': verificationEvidence.signature,
+            'account_id': decodeAccountId(accountId),
+            'username': username,
+            'phone_number_hash': phoneNumberHash,
+          }));
+
+      return (await signAndSendTransaction(call), null);
```

Comment:
Just typo, as you can see below we return result of `signAndsendTransaction` which return transaction hash
```suggestion
  /// Returns an (txHash, errorMessage) result.
  ///
  /// This method will attempt to obtain verifier evidence regarding the association between the accountId, userName and phoneNumber
  Future<(String?, String?)> newUser(
      String accountId, String username, String phoneNumber,
      {VerificationEvidence? verificationEvidence}) async {
    try {
      // Get verification evidence if not provided
      verificationEvidence ??= await getVerificationEvidence(
          accountId, username, phoneNumber,
          byPassToken: verificationBypassToken);

      // Failed to get verification evidence
      if (verificationEvidence == null) {
        return (null, 'NoVerifierEvidence');
      }

      // Verification failed
      if (verificationEvidence.verificationResult !=
          VerificationResult.verified) {
        return (null, verificationEvidence.verificationResult.toString());
      }

      final phoneNumberHash = hasher.hashString(phoneNumber);
      final call = MapEntry(
          'Identity',
          MapEntry('new_user', {
            'verifier_public_key':
                decodeAccountId(verificationEvidence.verifierAccountId!),
            'verifier_signature': verificationEvidence.signature,
            'account_id': decodeAccountId(accountId),
            'username': username,
            'phone_number_hash': phoneNumberHash,
          }));

      return (await signAndSendTransaction(call), null);
```

Reply:
We have an issue with tx hashes - in all tests, they stop to match - check the commented txHash code in most tests... it needs to be re-enabled.